projects/angular/src/data/datagrid/datagrid-page-size.ts

```diff
@@ -13,14 +13,15 @@ import { Page } from './providers/page';
   template: `
     <ng-content></ng-content>
     <div class="clr-select-wrapper">
-      <select [class.clr-page-size-select]="true" [(ngModel)]="page.size">
+      <select [id]="pageSizeOptionsId" [class.clr-page-size-select]="true" [(ngModel)]="page.size">
         <option *ngFor="let option of pageSizeOptions" [ngValue]="option">{{ option }}</option>
       </select>
     </div>
   `,
 })
 export class ClrDatagridPageSize {
   @Input('clrPageSizeOptions') pageSizeOptions: number[];
+  @Input('clrPageSizeOptionsId') pageSizeOptionsId = '';
```

Comment:
I just realized that this will cause the `id` attribute to default to `id=""` which isn't legal.

To fix this, you can inject `@Inject(UNIQUE_ID) publicuniquePageSizeOptionsId: string` and set `this. pageSizeOptionsId = uniquePageSizeOptionsId` in the constructor. This will ensure there is valid and unique default value and still allow a custom value to be set.

Reply:
Actually, you don't need to do the `@Inject(UNIQUE_ID)`. You can just initalize this property to `uniqueIdFactory()`.